tiledb/sm/storage_manager/context.cc

```diff
@@ -127,12 +100,46 @@ stats::Stats* Context::stats() const {
   return stats_.get();
 }
 
-Status Context::init_thread_pools(Config* const config) {
-  // If `config` is null, use a default-constructed config.
-  Config tmp_config;
-  if (config != nullptr)
-    tmp_config = *config;
+Status Context::init(const Config& config) {
+  RETURN_NOT_OK(init_loggers(config));
+
+  // Register stats.
+  stats::all_stats.register_stats(stats_);
+
+  // Create storage manager
+  storage_manager_ = new (std::nothrow)
+      tiledb::sm::StorageManager(&compute_tp_, &io_tp_, stats_.get(), logger_);
+  if (storage_manager_ == nullptr)
+    return logger_->status(Status_ContextError(
+        "Cannot initialize context Storage manager allocation failed"));
+
+  // Initialize storage manager
+  auto sm = storage_manager_->init(config);
 
+  return sm;
+}
+
+void Context::cleanup_storage_manager() {
```

Comment:
This is only called once, if I'm reading correctly. If so, it should just be the body of the destructor, rather than having a separate function for it.

Reply:
It's called in two locations: if the constructor fails (in `init(config)`) and in the destructor.

Reply:
Reading what's in the function, it's all stuff that either (1) should be in the `~StorageManager` or (2) could be solved with `unique_ptr`.

Needing to call anything like a `cleanup` is an antipattern that points out that whatever's being cleaned up isn't using RAII well. I think we could clean up `Context` here with a wrapper around `StorageManager` that does both of the above things so that `Context` can use a clean class. Then after this current work, a next unit of work can roll the wrapper into the class itself.

Reply:
@lums658 are you planning to address this or was it deferred to a story?